lang/cs/Org.Apache.REEF.Examples/MachineLearning/KMeans/KMeansDriverHandlers.cs

```diff
@@ -79,15 +80,23 @@ public KMeansDriverHandlers()
                 _executionDirectory); 
 
             _totalEvaluators = _partitionsNumber + 1;
-            _mpiDriver = new MpiDriver(Identifier, Constants.MasterTaskId, _fanOut, new AvroConfigurationSerializer());
-
-            _commGroup = _mpiDriver.NewCommunicationGroup(
-               Constants.KMeansCommunicationGroupName,
-               _totalEvaluators)
-                   .AddBroadcast(Constants.CentroidsBroadcastOperatorName, new BroadcastOperatorSpec<Centroids>(Constants.MasterTaskId, new CentroidsCodec()))
-                   .AddBroadcast(Constants.ControlMessageBroadcastOperatorName, new BroadcastOperatorSpec<ControlMessage>(Constants.MasterTaskId, new ControlMessageCodec()))
-                   .AddReduce(Constants.MeansReduceOperatorName, new ReduceOperatorSpec<ProcessedResults>(Constants.MasterTaskId, new ProcessedResultsCodec(), new KMeansMasterTask.AggregateMeans()))
+
+            IConfiguration mpiDriverConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindStringNamedParam<MpiConfigurationOptions.DriverId>(Identifier)
+                .BindStringNamedParam<MpiConfigurationOptions.MasterTaskId>(Constants.MasterTaskId)
+                .BindStringNamedParam<MpiConfigurationOptions.GroupName>(Constants.KMeansCommunicationGroupName)
+                .BindIntNamedParam<MpiConfigurationOptions.FanOut>(_fanOut.ToString(CultureInfo.InvariantCulture).ToString(CultureInfo.InvariantCulture))
+                .BindIntNamedParam<MpiConfigurationOptions.NumberOfTasks>(_totalEvaluators.ToString())
+                .Build();
+
+            _mpiDriver = TangFactory.GetTang().NewInjector(mpiDriverConfig).GetInstance<MpiDriver>();
```

Comment:
I actually thought about that way when I made the change. If I put MpiDriver as a parameter in the constructor of JobDriver, those bindings required by MdiDriver constructor would be part of the JobDriver config and will be serialized cross the bridge as part of the driver config file - file size would be bigger for download and serialization.  If we inject it inside JobDriver code, it would be just part of the code.  But if we think this data is small anyway, to let clients to set those bindings in one place, put MdiDriver in JobDriver constructor might be better.